consoleApp/src/main/java/com/infoshare/UserRegistration.java

```diff
@@ -0,0 +1,133 @@
+            }else{
+                System.out.println(REGISTRATION_ERROR);
+            }
+        }while (isGood == false);
+
+        return username;
+    }
+
+    private String getLocation(){
+        String location = "";
+        boolean isGood = false;
+        do{
+            System.out.println(REGISTRATION_LOCATION);
+            Scanner sc = new Scanner(System.in);
+            location = sc.nextLine();
+            if(isAlpha(location) == true){
+                isGood = true;
+            }else {
+                System.out.println(REGISTRATION_ERROR);
+            }
+        }while (isGood == false);
+        return location;
+    }
+
+    private String getPhoneNumber(){
+        String phone = "";
+        boolean isGood = false;
+        do{
+            System.out.println(REGISTRATION_PHONE_NUMBER);
+            Scanner sc = new Scanner(System.in);
+            phone = sc.nextLine();
+            if(isNumber(phone) == true){
+                isGood = true;
+            }else {
+                System.out.println(REGISTRATION_ERROR);
+            }
+        }while (isGood == false);
+        return phone;
+
+    }
+
+    private String getEmail(){
+        String email = "";
+        boolean isGood = false;
+        do{
+            System.out.println(REGISTRATION_EMAIL);
+            Scanner sc = new Scanner(System.in);
+            email = sc.nextLine();
+            if(isEmailValid(email) == true ){
+                isGood = true;
+            }else {
+                System.out.println(REGISTRATION_ERROR);
+            }
+        }while (isGood == false);
+        return email;
+    }
+
+    private String getHelpType(){
+        int helpType = 0;
+        boolean isGood = false;
+        TypeOfHelp[] typeOfHelpArray = TypeOfHelp.values();
+        do {
+            System.out.println(REGISTRATION_HELP_TYPE);
+            for (int i = 0; i < typeOfHelpArray.length; i++) {
+                System.out.println((i + 1) + " - " + typeOfHelpArray[i].getTypeOfHelp());
+            }
+            Scanner sc = new Scanner(System.in);
+            helpType = sc.nextInt();
+
+            if (helpType > 0 && helpType < typeOfHelpArray.length + 1) {
+                isGood = true;
+                helpType = helpType - 1;
+            } else {
+                System.out.println(REGISTRATION_ERROR);
+            }
+        }while (isGood == false);
+
+        return typeOfHelpArray[helpType].getTypeOfHelp();
+    }
+// [a-zA-Z]+[@][a-zA-Z]+[.][a-zA-Z]+
+    public boolean isAlpha(String name) {
+        return name.matches("[a-zA-Z]+");
+    }
+
+    public boolean isNumber(String number){
+        return number.matches("[0-9]+");
+    }
+
+    public boolean isEmailValid(String email) { return email.matches("^[A-Za-z0-9+_.-]+@[A-Za-z0-9.-]+$"); }
+
+    public void register(){
+        System.out.println(REGISTRATION_HEADER);
+
+        Volunteer newVolunteer = new Volunteer();
+
+        newVolunteer.setName(getVolunteerName());
```

Comment:
To przypisywanie bym wrzucił od razu do 119 i zamienił na jednolinijkowca

Reply:
zrobione :+1: